### ISSUE TITLE
pip install fails on fedora

### ISSUE THREAD

**Issue by jstraub** [facebookresearch/pyvrs#56]
## Instructions To Reproduce the 🐛 Bug:

I am simply trying to install vrs using pip install into a conda environment
```
$ pip install vrs  
```
I have tried version 1.0.3 and 1.0.4 and both fail for me.

This fails in the following way:

```
$ pip install vrs                                                                                                                                                                                                                               
Collecting vrs                                                                                                                                                                                                                                 
  Using cached vrs-1.0.4.tar.gz (19 kB)                                                                                                                                                                                                        
  Preparing metadata (setup.py) ... done                                                                                                                                                                                                       
Requirement already satisfied: numpy in /home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages (from vrs) (1.26.4)                                                                                                                     
Requirement already satisfied: typing in /home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages (from vrs) (3.7.4.3)                                                                                                                   
Requirement already satisfied: dataclasses in /home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages (from vrs) (0.6)                                                                                                                  
Building wheels for collected packages: vrs                                                                                                                                                                                                    
  Building wheel for vrs (setup.py) ... error                                                                                                                                                                                                  
  error: subprocess-exited-with-error                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
  × python setup.py bdist_wheel did not run successfully.                                                                                                                                                                                      
  │ exit code: 1                                                                                                                                                                                                                               
  ╰─> [74 lines of output]                                                                                                                                                                                                                     
      running bdist_wheel                                                                                                                                                                                                                      
      running build                                                                                                                                                                                                                            
      running build_py                                                                                                                                                                                                                         
      creating build                                                                                                                                                                                                                           
      creating build/lib.linux-x86_64-cpython-312                                                                                                                                                                                              
      creating build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                                        
      copying pyvrs/__init__.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                    
      copying pyvrs/base.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                        
      copying pyvrs/filter.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                      
      copying pyvrs/reader.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                      
      copying pyvrs/record.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                      
      copying pyvrs/slice.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                       
      copying pyvrs/utils.py -> build/lib.linux-x86_64-cpython-312/pyvrs                                                                                                                                                                       
      running build_ext                                                                                                                                                                                                                        
      CMake Warning:                                                                                                                                                                                                                           
        Ignoring extra path from command line:                                                                                                                                                                                                 
                                                                                                                                                                                                                                               
         "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6"                                                                                                                                                                      
                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                               
      CMake Error: The source directory "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6" does not appear to contain CMakeLists.txt.                                                                                                                                                                                                                                                                                                                                           
      Specify --help for usage, or press the help button on the CMake GUI.                                                                                                                                                                     
      Traceback (most recent call last):                                                                                                                                                                                                       
        File "<string>", line 2, in <module>                                                                                                                                                                                                   
        File "<pip-setuptools-caller>", line 34, in <module>                                                                                                                                                                                   
        File "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6/setup.py", line 124, in <module>                                                                                                                                  
          main()                                                                                                                                                                                                                               
        File "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6/setup.py", line 103, in main                                                                                                                                      
          setup(                                                                                                                                                                                                                               
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/__init__.py", line 104, in setup                                                                                                                       
          return distutils.core.setup(**attrs)                                                                                                                                                                                                 
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                 
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 184, in setup                                                                                                                
          return run_commands(dist)                                                                                                                                                                                                            
                 ^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                            
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 200, in run_commands                                                                                                         
          dist.run_commands()                                                                                                                                                                                                                  
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands                                                                                                         
          self.run_command(cmd)                                                                                                                                                                                                                
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/dist.py", line 967, in run_command                                                                                                                     
          super().run_command(command)                                                                                                                                                                                                         
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 988, in run_command                                                                                                          
          cmd_obj.run()                                                                                                                                                                                                                        
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/wheel/bdist_wheel.py", line 368, in run                                                                                                                           
          self.run_command("build")                                                                                                                                                                                                            
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command                                                                                                           
          self.distribution.run_command(command)                                                                                                                                                                                               
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/dist.py", line 967, in run_command                                                                                                                     
          super().run_command(command)                                                                                                                                                                                                         
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 988, in run_command                                                                                                          
          cmd_obj.run()                                                                                                                                                                                                                        
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 132, in run                                                                                                         
          self.run_command(cmd_name)                                                                                                                                                                                                           
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command                                                                                                           
          self.distribution.run_command(command)                                                                                                                                                                                               
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/dist.py", line 967, in run_command                                                                                                                     
          super().run_command(command)                                                                                                                                                                                                         
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 988, in run_command                                                                                                          
          cmd_obj.run()                                                                                                                                                                                                                        
        File "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6/setup.py", line 64, in run                                                                                                                                        
          super().run()                                                                                                                                                                                                                        
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/command/build_ext.py", line 91, in run                                                                                                                 
          _build_ext.run(self)                                                                                                                                                                                                                 
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run                                                                                                     
          self.build_extensions()                                                                                                                                                                                                              
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 479, in build_extensions                                                                                                                                                                                                                                                                                                                                       
          self._build_extensions_serial()                                                                                                                                                                                                      
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 505, in _build_extensions_serial                                                                                                                                                                                                                                                                                                                               
          self.build_extension(ext)                                                                                                                                                                                                            
        File "/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6/setup.py", line 91, in build_extension                                                                                                                            
          subprocess.check_call(                                                                                                                                                                                                               
        File "/home/jstraub/miniconda3/envs/efm/lib/python3.12/subprocess.py", line 413, in check_call                                                                                                                                         
          raise CalledProcessError(retcode, cmd)                                                                                                                                                                                               
      subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6', '-DCMAKE_BUILD_TYPE=Release', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-hzt24sw9/vrs_81da9d1284bf48abafb0895443be8ac6/build/lib.linux-x86_64-cpython-312/', '-DPYTHON_EXECUTABLE=/home/jstraub/miniconda3/envs/efm/bin/python', '-GCodeBlocks', '-GNinja']' returned non-zero exit status 1.
      [end of output]                                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
  note: This error originates from a subprocess, and is likely not a problem with pip.                                                                                                                                                         
  ERROR: Failed building wheel for vrs                                                                                                                                                                                                         
  Running setup.py clean for vrs                                                                                                                                                                                                               
Failed to build vrs                                                                                                                                                                                                                            
ERROR: Could not build wheels for vrs, which is required to install pyproject.toml-based projects   
```

**Comment by ddetone** [comment]
also hitting this on fedora 39

**Comment by SeaOtocinclus** [comment]
It seems you are using python 3.12 and vrs python is build up to python 3.11 😉
https://pypi.org/project/vrs/1.0.4/#files

**Comment by ddetone** [comment]
SOLVED for me by downgrading python from 3.12 to 3.11
